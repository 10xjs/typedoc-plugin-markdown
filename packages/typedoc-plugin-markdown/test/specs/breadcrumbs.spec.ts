import * as Handlebars from 'handlebars';
import { Reflection } from 'typedoc';
import { TestApp } from '../test-app';

describe(`Breadcrumbs:`, () => {
  let testApp: TestApp;
  let moduleReflection: Reflection;
  let classReflection: Reflection;

  beforeAll(() => {
    testApp = new TestApp(['breadcrumbs.ts']);
  });

  describe(`(with readme)`, () => {
    beforeAll(() => {
      testApp.bootstrap();
      moduleReflection = testApp.findModule('breadcrumbs');
      classReflection = testApp.project.findReflectionByName('Breadcrumbs');
    });

    test(`should compile README breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: testApp.project,
          model: testApp.project,
          url: 'README.md',
        }),
      ).toMatchSnapshot();
    });

    test(`should compile entryPoint (globals) breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: testApp.project,
          model: testApp.findEntryPoint(),
          url: 'globals.md',
        }),
      ).toMatchSnapshot();
    });

    test(`should compile module breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: testApp.project,
          model: moduleReflection,
          url: moduleReflection.url,
        }),
      ).toMatchSnapshot();
    });
    test(`should compile class breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: testApp.project,
          model: classReflection,
          url: classReflection.url,
        }),
      ).toMatchSnapshot();
    });
  });
  describe(`(without readme)`, () => {
    beforeAll(() => {
      testApp.bootstrap({ readme: 'none' });
      moduleReflection = testApp.findModule('breadcrumbs');
      classReflection = testApp.project.findReflectionByName('Breadcrumbs');
    });

    test(`should not compile entryPoint (globals) breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: testApp.project,
          model: testApp.findEntryPoint(),
          url: 'globals.md',
        }),
      ).toBeNull();
    });

    test(`should compile module breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: testApp.project,
          model: moduleReflection,
          url: moduleReflection.url,
        }),
      ).toMatchSnapshot();
    });
    test(`should compile class breadcrumbs'`, () => {
      expect(
        Handlebars.helpers.breadcrumbs.call({
          project: testApp.project,
          model: classReflection,
          url: classReflection.url,
        }),
      ).toMatchSnapshot();
    });
  });
});
