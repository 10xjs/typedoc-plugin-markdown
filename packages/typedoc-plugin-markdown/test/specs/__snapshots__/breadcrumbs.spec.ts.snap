// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs: (with readme) should compile README breadcrumbs' 1`] = `"README / [Globals](../globals.md)"`;

exports[`Breadcrumbs: (with readme) should compile class breadcrumbs' 1`] = `"[README](../README.md) / [Globals](../globals.md) / [\\"breadcrumbs\\"](../modules/_breadcrumbs_.md) / Breadcrumbs"`;

exports[`Breadcrumbs: (with readme) should compile entryPoint (globals) breadcrumbs' 1`] = `"[README](../README.md) / Globals"`;

exports[`Breadcrumbs: (with readme) should compile module breadcrumbs' 1`] = `"[README](../README.md) / [Globals](../globals.md) / \\"breadcrumbs\\""`;

exports[`Breadcrumbs: (without readme) should compile class breadcrumbs' 1`] = `"[Globals](../README.md) / [\\"breadcrumbs\\"](../modules/_breadcrumbs_.md) / Breadcrumbs"`;

exports[`Breadcrumbs: (without readme) should compile module breadcrumbs' 1`] = `"[Globals](../README.md) / \\"breadcrumbs\\""`;
